variables.tf

```diff
@@ -92,6 +92,12 @@ variable "user_data" {
   type        = string
 }
 
+variable "use_boot_volume_key_as_default" {
+  description = "Set to true to use boot volume encryption key as the default for all volumes."
+  type        = bool
+  default     = false
```

Comment:
@toddgiguere @Aashiq-J should we default this to true or false?

Based on the results of the upgrade test, was I think this would of defaulted to `true` before this change was added? Not saying that is what we need to stick to, but Ill need to know for the release notes what to say here to prevent breaking change for people upgrading

Reply:
@ocofaigh , @toddgiguere ,
I guess, we should keep it `false` to not use boot_volume key for all the volumes created in the newer releases.
But if a customer wants to prevent the breaking change they can set it to true manually.

Reply:
yes I'm with that - we can document it in the release notes.

Reply:
@Aashiq-J @toddgiguere how are we going to handle this in the landing zone DA? I know we can't go to v4 of this module until we have migration support, but when we do, will we need to set this to `true` to prevent a breaking change for existing customers?

Reply:
@ocofaigh the VSI pattern does not allow for specifying any additional block volumes, so from that point it won't matter.

The VSI map though does allow for volumes and allows for encryption keys for those volumes, and also has `kms_encryption_enabled` defaulted to "true" which would have meant that the boot key would have been used for all volumes.

Based on that I would say that to avoid destruction of existing attached volumes, you would want to default the new option to "true".